ekncontent/ekncontent/eknc-utils.c

```diff
@@ -14,6 +16,86 @@
  * Utility functions for the Endless Knowledge Content library.
  */
 
+/**
+ * eknc_utils_append_gparam_from_json_node:
+ *
+ * For use instantiating a GObject from json data. Marshals the contents
+ * of a JsonNode into a GParameter matching the GParamSpec pspec. Appends
+ * that GParameter to a GArray, for eventual use with g_object_newv.
+ */
+void
+eknc_utils_append_gparam_from_json_node (JsonNode *node,
+                                         GParamSpec *pspec,
+                                         GArray *params)
+{
+  g_autoptr(GError) error = NULL;
+  GParameter param = { NULL, { 0, }};
+
+  if (!node)
+    return;
+
+  param.name = pspec->name;
+  g_value_init (&param.value, pspec->value_type);
+
+  if (G_IS_PARAM_SPEC_VARIANT (pspec))
+    {
+      GVariantType *type = G_PARAM_SPEC_VARIANT (pspec)->type;
+      GVariant *variant = json_gvariant_deserialize (node,
+                                                     g_variant_type_peek_string (type),
+                                                     &error);
+      g_value_set_variant (&param.value, variant);
+    }
+  else if (JSON_NODE_HOLDS_VALUE (node))
+    {
+      GValue value = { 0, };
+      GType type = json_node_get_value_type (node);
+
+      // For some reason some of our integer properties are stored in json as
+      // strings. Try to parse if we hit this case.
```

Comment:
I believe this is because ImageObject, which our schemas are based on, have width/height defined as a "Distance": https://schema.org/ImageObject which is supposed to be a unit-ful value like "500px" or "5 inches". We don't use any of that, but width/height are indeed defined as strings in our object models: https://github.com/endlessm/endless-content-factory/blob/master/src/eos-knowledge-db-build/database_build/packer/models/media_object.py#L17-L18